### PR TITLE
mimic: osd/PeeringState: do not complain about past_intervals constrained by oldest epoch

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/41502

---

backport of https://github.com/ceph/ceph/pull/29747
parent tracker: https://tracker.ceph.com/issues/39546

this backport was staged using https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
@neha-ojha Thanks - I somehow neglected to add that comment even though I knew it was needed.

Also there was a make check failure first time around, but I didn't get a chance to look at it - let's see if it happens a second time.

File: src/osd/PG.cc [hunk 1]
```diff
@@ -896,11 +896,15 @@ bool PG::needs_backfill() const
 
 void PG::check_past_interval_bounds() const
 {
+  auto oldest_epoch = osd->get_superblock().oldest_map;
   auto rpib = get_required_past_interval_bounds(
     info,
-    osd->get_superblock().oldest_map);
+    oldest_epoch)
```
Comment (tchaikov) [inline src/osd/PG.cc]:
```suggestion
    oldest_epoch);
```

Comment (smithfarm) [reply src/osd/PG.cc]: thanks

**Comment by yuriw** [comment]
[wip-yuri2-testing-2019-09-30-1548](https://trello.com/c/v1bxeK8n/698-wip-yuri2-testing-2019-09-30-1548)

**Pull request merged**